### ISSUE TITLE
Icons are low-res

### ISSUE THREAD

**Issue by danieleds** [notepadqq/notepadqq#768]
Our icon theme is low-res and looks very bad on high DPI screens.

On Ubuntu, we see the problem for the icons not associated with a system theme:

<img width="529" alt="screen shot 2018-09-22 at 11 21 35 2" src="https://user-images.githubusercontent.com/402652/45915719-3af0e500-be5a-11e8-96fd-9641e49759b2.png">

on macOS, all the icons have this problem:
<img width="501" alt="screen shot 2018-09-22 at 11 25 02" src="https://user-images.githubusercontent.com/402652/45915723-404e2f80-be5a-11e8-909c-cb833714f66f.png">

A new icon theme would be nice to have.

(if you're not on a high DPI screen, click to see the unscaled images)

**Comment by danieleds** [comment]
Some ideas that I'm considering:

https://material.io/tools/icons/?style=outline

https://feathericons.com/?ref=producthunt

https://icomoon.io/#preview-free

https://fontawesome.com/icons?d=gallery&m=free

**Comment by JuBan1** [comment]
One issue with monochrome icons is that they'll be very hard to see on dark themed desktops.

I'm surprised some desktop themes comes without save-all, document-close, or document-close-all. Those aren't exactly uncommon.

**Comment by danieleds** [comment]
> One issue with monochrome icons is that they'll be very hard to see on dark themed desktops.

Yes, that's something that we should definitely keep in mind. Actually, Google offers both white and dark versions, so we can decide to switch them automatically if there isn't enough contrast.

Or, given the fact that they're monochrome, we could even invert the colors by code. Either on user's choice, or by checking the contrast.

**Comment by Teklad** [comment]
Should be able to do this easily by adding rgb values together, and some pseudo-code assuming we base it on widget background color:
```cpp
const int contrast = (r + g + b) / 3;
if (contrast > 160) { DarkTheme; }
else if (contrast < 96) { LightTheme; }
else { DefaultTheme; }
```

Since all the images you chose are gray scale this is probably one of the better ways to go about it.

**Comment by JuBan1** [comment]
FYI, this code works:

```
    // Load the image and create a color mask (in this case, from black)
    QIcon icon("/path/to/image.svg");
    auto pm = icon.pixmap(QSize(size,size));
    auto mask = pm.createMaskFromColor(Qt::black, Qt::MaskOutColor);
 
    // Create the new pixmap and paint using the color mask.
    QPixmap pix = QPixmap(size, size);
    pix.fill(Qt::transparent);
    QPainter painter(&pix);
    painter.setPen(Qt::red);
    painter.drawPixmap(pm.rect(), mask, mask.rect());
```

Result:

![screenshot_20180928_185300](https://user-images.githubusercontent.com/4319621/46222159-c03f3280-c34f-11e8-96b6-52e759c67d10.png)

**Comment by danieleds** [comment]
Fixed in 25f98a896a86a1f7832db19dfd5e658cc3fbf0c0
